django/db/backends/postgresql/compiler.py

```diff
@@ -43,7 +45,7 @@ def assemble_as_sql(self, fields, value_rows):
         db_types = [field.db_type(self.connection) for field in fields]
         # Abort if any of the fields are arrays as UNNEST indiscriminately
         # flatten them instead of reducing their nesting by one.
-        if any(db_type.endswith("[]") for db_type in db_types):
+        if any(re.search(r".+\[\d*\]$", db_type) for db_type in db_types):
```

Comment:
Since the regular expression is used in the list comprehension loop, might it be better to compile it in advance ?
https://docs.python.org/3/library/re.html#re.compile